ElYrZR34 Register users and authorizations

mansurissa opened the pull request:
#### What does this PR do?
> Registering users and sending them emails
#### Description of Task to be completed
- Create user model and migration
- create seed users
- registering users
- sending verfication and comfirmation email

#### How should this be manually tested?
1. clone this repo
1. run `npm install` 
1. create `.env` file and put configs in as in `.env.example` 
1. run `npm run dev` or `npm start`


####  Relevant trello card?
ElYrZR34

mansurissa committed:
ch-setup(initial setup): setting up project

- install express
- install dotenv
- making test route

mansurissa committed:
ch-setup(initial setup): setting up project

- install express
- install dotenv
- making test route

GuevaraMarius committed:
feat(authentication):signin

- create user model and migration
- create seeder users
- hash password
- sign in
- asign token to signed in user

chore:(eslint): configuringneslint

- install and configure eslint
- using airbnb base style

chore: connect-to-db
- creating seeds
- creating migration
- user model

 bg-fixing some model error

ft: signin

- registering user
- validations with join

 hashing pass2word before saving in db

feat(authenticatiom):sign in

- login for user
- signing token to signed in user

GuevaraMarius committed:
feat(authentication):signin

- create user model and migration
- create seeder users
- hash password
- sign in
- asign token to signed in user

chore:(eslint): configuringneslint

- install and configure eslint
- using airbnb base style

chore: connect-to-db
- creating seeds
- creating migration
- user model

 bg-fixing some model error

ft: signin

- registering user
- validations with join

 hashing pass2word before saving in db

feat(authenticatiom):sign in

- login for user
- signing token to signed in user

KundwaBruno committed:
Using google cloud transalte api for supporting multi languages

Changed to ES6 syntax

Fix issue #15

Fixed Issue_#15'

issue_#15

Fixed issue #15 and i18n setup

xUjJi1sq Chore Support multiple languages

ch-CRUD-role

CREATE Roles
READ Roles
UPDATE Roles
DELETE Roles

ch-permissions-middleware

Fix issue #15

Fixed Issue_#15'

issue_#15

Fixed issue #15 and i18n setup

xUjJi1sq Chore Support multiple languages

ch-setup(initial setup): setting up project

- install express
- install dotenv
- making test route

feat(authentication):signin

- create user model and migration
- create seeder users
- hash password
- sign in
- asign token to signed in user

chore:(eslint): configuringneslint

- install and configure eslint
- using airbnb base style

chore: connect-to-db
- creating seeds
- creating migration
- user model

 bg-fixing some model error

ft: signin

- registering user
- validations with join

 hashing pass2word before saving in db

feat(authenticatiom):sign in

- login for user
- signing token to signed in user

ch-setup(initial setup): setting up project

- install express
- install dotenv
- making test route

feat(authentication):signin

- create user model and migration
- create seeder users
- hash password
- sign in
- asign token to signed in user

chore:(eslint): configuringneslint

- install and configure eslint
- using airbnb base style

chore: connect-to-db
- creating seeds
- creating migration
- user model

 bg-fixing some model error

ft: signin

- registering user
- validations with join

 hashing pass2word before saving in db

feat(authenticatiom):sign in

- login for user
- signing token to signed in user

feat(register) registring users

- random password
- sending email(not yet done)

 generating random password

 sending email

verify endpoint

 creating midleware for checking token

feat:(register): registering users

- registering user
- sending email after creating user
- sending comfirmation email after verifying account

ElYrZR34

KundwaBruno committed:
Using google cloud transalte api for supporting multi languages

Changed to ES6 syntax

KundwaBruno committed:
Fix issue #15

Fixed Issue_#15'

issue_#15

Fixed issue #15 and i18n setup

xUjJi1sq Chore Support multiple languages

41hulk committed:
ch-CRUD-role

CREATE Roles
READ Roles
UPDATE Roles
DELETE Roles

ghost commented:
## DeepCode's analysis on [#8fe11a](https://www.deepcode.ai/app/gh/atlp-rwanda/incredible-phantom-backend/ded2ba2399cc7a31a4458bf52b1abb247b86ec4f/atlp-rwanda/incredible-phantom-backend/8fe11a2d860ba71567cfad07706d5d9e44b522b3/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) found:

- :information_source: **1** minor issue. :point_down:
## Top issues

<table>
  <thead>
    <tr>
      <th align="left">Description</th>
      <th align="left">Example fixes</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td width="77%">Missing JSDoc for parameter 'models'. Occurrences: <ul><li><a href="https://github.com/atlp-rwanda/incredible-phantom-backend/blob/8fe11a2d860ba71567cfad07706d5d9e44b522b3/src/database/models/user.js#L4">user.js:4</a></li></ul></td>
      <td width="23%"><a href="https://www.deepcode.ai/app/gh/atlp-rwanda/incredible-phantom-backend/ded2ba2399cc7a31a4458bf52b1abb247b86ec4f/atlp-rwanda/incredible-phantom-backend/8fe11a2d860ba71567cfad07706d5d9e44b522b3/pr/_/%2Fsrc%2Fdatabase%2Fmodels%2Fuser.js/javascript%2Feslint%2Fvalid-jsdoc/code/?utm_source=gh_review&">:wrench: Example fixes</a></td>
    </tr>
  </tbody>
</table>


#### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/atlp-rwanda/incredible-phantom-backend/ded2ba2399cc7a31a4458bf52b1abb247b86ec4f/atlp-rwanda/incredible-phantom-backend/8fe11a2d860ba71567cfad07706d5d9e44b522b3/pr/_/%2F/code/?utm_source=gh_review&c=0&w=0&i=1&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=atlp-rwanda)

41hulk committed:
ch-permissions-middleware

41hulk committed:
Fix-issue-package-lock

41hulk committed:
feat(Roles & Permissions): Setting Roles and Permissions

-Admin can create,update,delete roles
-Admin can set and update Roles's permission

[Delivers #P4Z442wG]
- Ntare Guy -

mansurissa committed:
feat(authorization):Register and routes protection

- register user
- get all users
- create role
- protecting routes
- permissions according to the role

 adding travis config for database and tests

 adding documentation

samuel776 reviewed (commented):
Hey Issa, for sure you did a great job, if you allow me i can suggest you to work on the conficts and if possible you should add the screenshot showing registration success in postman. thank you
